wave_matching/tests/icp_tests.cpp

```diff
@@ -0,0 +1,80 @@
+// Tests for wrapper
+
+#include <gtest/gtest.h>
+#include <pcl/io/pcd_io.h>
+#include <Eigen/Dense>
+#include <iostream>
+
+#include "wave/matching/ICP.hpp"
+
+bool isApprox(const Eigen::Affine3d& first,
+              const Eigen::Affine3d& second,
+              double TOL) {
+    for (uint8_t iii = 0; iii < 4; ++iii) {
```

Comment:
1 last thing, admittedly stupid detail its up to you, but consider using `i++` instead of `++i`. While both are perfectly valid I rarely (if ever) see `++i`. 

Once you're outside the norm, subtle off by one errors creep in. For example these two loops do different things.

```
int i = 0;
int j = i;
while(j < 5) {
    printf("%d", i);
    j = ++i;
}
// prints 1 2 3 4

int i = 0;
int j = i;
while(j < 5) {
    printf("%d", i);
    j = i++;
}
// prints 0 1 2 3 4 5 
```

http://ideone.com/U3Wz2K